CONTRIBUTING.md

```diff
@@ -0,0 +1,61 @@
+# Contributing to theupdateframework.io
+We're excited that you're interested in contributing to the project! 🎉 Your contributions are highly valued and welcome. 🎉
+
+## Issues & Pull Requests
+
+### Creating an Issue
+
+Before **creating** an Issue i.e for `bug`/`features`/`improvements` please follow these steps:
+
+
+1.Check Existing Issues: Search to see if the issue already exists.
+
+2.Provide Context: If it doesn't exist, create a new issue with as much detail as possible. Make sure to select the correct issue type (e.g. bug,documentation,feature).
+
+3.Express Interest: If you want to work on the issue after it's been reviewed, mention this in your issue description.
+
+### Working on an Issue
+
+Before starting work on an existing issue, please follow these steps:
+
+1. **Request Assignment**: Comment on the issue asking for it to be assigned to you.
+2. **Prepare for Assignment**:
+    - Confirm that you have read the `CONTRIBUTING.md`.
+    - Ensure you have a functional development environment (you can build and run the project).
+    - Describe your intended approach to solving the issue.
+    - Include these points in one or more comments.
+3. **Start Work**: Begin working on the issue only after it has been assigned to you.
+4. **Avoid Conflicts**: Only start working on the issue and open a Pull Request after it has been assigned to you. This prevents confusion, duplicate work, and potential conflicts among contributors.
+5. **Reference the Issue**: In your Pull Request, reference the issue (e.g., `This PR fixes #1234`). This ensures the issue is automatically closed when your Pull Request is merged.
```

Comment:
Hi @Ayush9026, I think we don't have functionality of auto closing the issue. [See ](https://github.com/theupdateframework/theupdateframework.io/pull/39)even Referencing the issue in PR, we have to close it manually [see ](https://github.com/theupdateframework/theupdateframework.io/issues/37). So IMO we should remove this part. 

@lukpueh @joshuagl correct me if i am wrong 🙏

Reply:
@Ayush9026 you did a wonderful work, I've just a small suggestion, should we add testing guidelines too, on how to test changes locally same as [python-tuf CONTRIBUTING guideline.](https://theupdateframework.readthedocs.io/en/latest/CONTRIBUTING.html). Steps can be this:
Install Project Dependencies:

-  Execute the following command to install the project's dependencies locally:
   ` $ yarn install`
    
-  Start the Hugo development server to preview the website locally by running:
   `$ hugo server`
  
-   Once started, the website will be accessible at: http://localhost:1313/    by default.

Reply:
Hi @h4l0gen,

Thanks 🙏 for letting me know. If auto-closing isn't supported, I'll remove that part once @lukpueh and @joshuagl review it.